From 90733966b800d3d2ac084b91dc5b0a408fbb052e Mon Sep 17 00:00:00 2001
From: Etienne Carriere <etienne.carriere@foss.st.com>
Date: Mon, 2 Sep 2024 17:32:20 +0200
Subject: [PATCH 1179/1695] drivers: stm32_ltdc: remove
 stm32_pinctrl_set_secure_cfg()

Remove use of stm32_pinctrl_set_secure_cfg() now that the pinctrl secure
and non-secure configuration is part of the alternate firewall
configurations retrieved from the DT.

Upstream-Status: Inappropriate [ST downstream https://github.com/STMicroelectronics/optee_os.git e08874e6b49941a3894b7d2b8b8fc5f0f8d7c10b drivers: stm32_ltdc: remove stm32_pinctrl_set_secure_cfg()]
Change-Id: Ic10d5dddcb6b9f55ddcaeba324f2f712e21a7ca6
Signed-off-by: Etienne Carriere <etienne.carriere@foss.st.com>
Reviewed-on: https://gerrit.st.com/c/mpu/oe/optee/optee_os/+/394352
Domain-Review: Yann GAUTIER <yann.gautier@foss.st.com>
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
Reviewed-by: Thomas BOURGOIN <thomas.bourgoin@foss.st.com>
ACI: CIBUILD <MDG-smet-aci-builds@list.st.com>
---
 core/drivers/stm32_ltdc.c | 3 ---
 1 file changed, 3 deletions(-)

diff --git a/core/drivers/stm32_ltdc.c b/core/drivers/stm32_ltdc.c
index 8b3991205..d0bdb7d11 100644
--- a/core/drivers/stm32_ltdc.c
+++ b/core/drivers/stm32_ltdc.c
@@ -157,9 +157,6 @@ static TEE_Result stm32_ltdc_init(void *device)
 		goto err;
 	}
 
-	if (ldev->pinctrl)
-		stm32_pinctrl_set_secure_cfg(ldev->pinctrl, true);
-
 	/* Force the LTDC to secure access */
 	ret = firewall_set_alternate_conf(ldev->sec_conf);
 	if (ret)
-- 
2.47.2

